modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/DefaultArchiveDeployer.java

```diff
@@ -172,6 +175,23 @@ public T deployPackagedArtifact(URI artifactAchivedUri) throws DeploymentExcepti
     }
   }
 
+  private void undeployArtifactIfItsAPatch(T artifact) {
+    Optional<T> foundMatchingArtifact = artifacts.stream().filter(deployedArtifact -> {
+      BundleDescriptor deployedBundleDescriptor = deployedArtifact.getDescriptor().getBundleDescriptor();
+      BundleDescriptor artifactBundleDescriptor = artifact.getDescriptor().getBundleDescriptor();
+      if (deployedBundleDescriptor.getGroupId().equals(artifactBundleDescriptor.getGroupId()) &&
+          deployedBundleDescriptor.getArtifactId().equals(artifactBundleDescriptor.getArtifactId())) {
+        MuleVersion deployedVersion = new MuleVersion(deployedBundleDescriptor.getVersion());
+        MuleVersion artifactVersion = new MuleVersion(artifactBundleDescriptor.getVersion());
+        if (artifactVersion.sameBaseVersion(deployedVersion)) {
+          return true;
```

Comment:
What's the expectation if I have 1.0.3 deployed and drop 1.0.2? This suggests 1.0.3 will be taken down. Add a test case for that scenario (whether that is correct or not).